pkg/storageincentives/agent.go

```diff
@@ -316,7 +316,7 @@ func (a *Agent) handleClaim(ctx context.Context, round uint64) (bool, error) {
 		return false, nil
 	}
 
-	a.state.SetLastPlayedRound(round)
+	a.state.SetLastSelectedRound(round)
```

Comment:
Personally, "LastSelectedRound" sounds like it should be set when the node's neighborhood is selected, like right at the start of "handleSample"?  Even before the checks to see if we will be playing that round.

Reply:
Well, considering the fact that  'neighbourhood chosen' together with the round is logged to the user after the checks, just before making the sample, it made sense to set the `LastSelectedRound` there. What do you think?

Reply:
But yeah, should be swapped with `LastPlayedRound`

Reply:
> But yeah, should be swapped with `LastPlayedRound`

Only if SetSelectedRound is moved up to the beginning before the checks.   The neighborhood is selected for a round even if the node chooses not to play for some reason.   I'd figure that's the primary difference between "Selected" and "Played".

Reply:
right, will amend, but wouldn't that make the positioning of the log message `a.logger.Info("neighbourhood chosen", "round", round)` inaccurate? Since all the checks have to pass before its logged?